public/components/event_analytics/explorer/events_views/trace_block/trace_block.tsx

```diff
@@ -22,7 +22,7 @@ interface props {
 }
 
 export const TraceBlock = ({ http, hit, logTraceId }: props) => {
-  if (logTraceId === '' || !isValidTraceId(logTraceId)) {
+  if ((!hit.traceID || hit.traceID.length === 0) && (logTraceId === '' || !isValidTraceId(logTraceId))){
```

Comment:
Regarding your first point, I think you're right - will fix. Regarding your second point I don't think so... shouldn't we assume abstraction between how data is stored? Otherwise we will be handling a lot of edge cases, as long as the UI doesn't break if its not stored in the way we expect we just don't provide the functionality.